components/DateTimePicker.js

```diff
@@ -0,0 +1,161 @@
+import React, { PureComponent } from 'react';
+import RNCDateTimePicker from '@react-native-community/datetimepicker';
+import autoBindReact from 'auto-bind/react';
+import PropTypes from 'prop-types';
+import { Platform } from 'react-native';
+import Modal from 'react-native-modal';
+import { connectStyle } from '@shoutem/theme';
+import { Icon } from './Icon';
+import { Text } from './Text';
+import { TouchableOpacity } from './TouchableOpacity';
+import { View } from './View';
+import { Button } from './Button';
+
+const isIos = Platform.OS === 'ios';
+
+class DateTimePicker extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    autoBindReact(this);
+
+    this.state = {
+      showPicker: false,
+      value: props.value,
+    };
+  }
+
+  componentDidUpdate() {
+    if (!isIos) {
+      const { value } = this.props;
+
+      this.setState({ value });
+    }
+  }
+
+  handleValueChanged(event, value) {
+    if (isIos) {
+      return this.setState({ value });
+    }
+
+    if (event.type === 'dismissed') {
+      return this.setState({ showPicker: false });
+    }
+
+    const { onValueChanged } = this.props;
+
+    this.setState({ showPicker: false });
+    return onValueChanged(value);
+  }
+
+  handleShowPicker() {
+    this.setState({ showPicker: true });
+  }
+
+  handleHidePicker() {
+    this.setState({ showPicker: false });
+  }
+
+  handleConfirmPress() {
+    const { onValueChanged } = this.props;
+    const { value } = this.state;
+
+    onValueChanged(value);
+    this.handleHidePicker();
+  }
+
+  render() {
+    const {
+      confirmButtonText,
+      is24Hour,
+      mode,
+      textValue,
+      style,
+      ...otherProps
+    } = this.props;
+    const { showPicker, value } = this.state;
+
+    return (
+      <View styleName="horizontal">
+        <TouchableOpacity
+          styleName="flexible md-gutter"
+          style={style.textContainer}
+          onPress={this.handleShowPicker}
+        >
+          <Text> {textValue}</Text>
+        </TouchableOpacity>
+        <TouchableOpacity
+          onPress={this.handleShowPicker}
+          styleName="h-center v-center"
+          style={style.buttonContainer}
+        >
+          <Icon name="drop-down" style={style.icon} />
+        </TouchableOpacity>
+        {isIos && (
+          <Modal
```

Comment:
Not a request change, but personally I like to add props such as `swipeDirection` and `onSwipeComplete` to modal to allow users dismissing modal via swiping up or down. 
This is more of a UX change so it's up to your preference.

Reply:
This looks super cool, defo UX improvement :D 
Thx!

https://user-images.githubusercontent.com/57353746/126150762-ba44f37b-290d-4e1b-8001-2378b10507d0.mov